src/Microsoft.PowerShell.ConsoleHost/host/msh/Executor.cs

```diff
@@ -320,11 +322,346 @@ internal Collection<PSObject> ExecuteCommand(string command, out Exception excep
+                }
+
+                return AstVisitAction.Continue;
+            }
+
+            public override AstVisitAction VisitCommand(CommandAst commandAst)
+            {
+                if (commandAst.InvocationOperator == TokenKind.Dot)
+                {
+                    ThrowError(
+                        new ImplicitRemotingBatchingNotSupportedException(
+                            ConsoleHostStrings.ImplicitBatchDotSourcingNotSupported),
+                        commandAst);
+                }
+
+                /*
+                // Up front checking ensures that we have a simple script block,
+                // so we can safely assume that the parents are:
+                //     * a PipelineAst
+                //     * a NamedBlockAst (the end block)
+                //     * a ScriptBlockAst (the ast we're comparing to)
+                // If that isn't the case, the conversion isn't allowed.  It
+                // is also safe to assume that we have at least 3 parents, a script block can't be simpler.
+                if (commandAst.Parent.Parent.Parent != ScriptBeingConverted)
+                {
+                    ThrowError(
+                        new ImplicitRemotingBatchingNotSupportedException(
+                            "CantConvertWithCommandInvocations not supported"),
+                        commandAst);
+                }
+                */
+
+                if (commandAst.CommandElements[0] is ScriptBlockExpressionAst)
+                {
+                    ThrowError(
+                        new ImplicitRemotingBatchingNotSupportedException(
+                            ConsoleHostStrings.ImplicitBatchScriptBlockNotSupported),
+                        commandAst);
+                }
+
+                var commandName = commandAst.GetCommandName();
+                if (commandName != null)
+                {
+                    Commands.Add(commandName);
+                }
+
+                return AstVisitAction.Continue;
+            }
+
+            public override AstVisitAction VisitMergingRedirection(MergingRedirectionAst redirectionAst)
+            {
+                if (redirectionAst.ToStream != RedirectionStream.Output)
+                {
+                    ThrowError(
+                        new ImplicitRemotingBatchingNotSupportedException(
+                            ConsoleHostStrings.ImplicitBatchMergeRedirectNotSupported),
+                        redirectionAst);
+                }
+
+                return AstVisitAction.Continue;
+            }
+
+            public override AstVisitAction VisitFileRedirection(FileRedirectionAst redirectionAst)
+            {
+                ThrowError(
+                    new ImplicitRemotingBatchingNotSupportedException(
+                        ConsoleHostStrings.ImplicitBatchFileRedirectNotSupported),
+                    redirectionAst);
+
+                return AstVisitAction.Continue;
+            }
+
+            /*
```

Comment:
Please remove commented block.

Reply:
Since this is experimental, I want to leave it in for now.  We may need some kind of script block restriction.